vignettes/visits_periods.Rmd

```diff
@@ -0,0 +1,273 @@
+---
+title: "Visit and Period Variables"
+output:
+  rmarkdown::html_vignette:
+vignette: >
+  %\VignetteIndexEntry{Visit and Period Variables}
+  %\VignetteEngine{knitr::rmarkdown}
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+# Introduction
+
+The derivation of visit variables like `AVISIT`, `AVISITN`, `AWLO`, `AWHI`, ...
+or period, subperiod, or phase variables like `APERIOD`, `TRT01A`, `TRT02A`,
+`ASPER`, `PHSDTM`, `PHEDTM`, ... is highly study-specific. Therefore admiral can
+not provide functions which derive these variables. However, for common
+scenarios like visit assignments based on time windows or deriving BDS period
+variables from ADSL period variables functions are provided which support those
+derivations.
+
+## Required Packages
+
+The examples of this vignette require the following packages.
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(tibble)
+library(dplyr)
+library(lubridate)
+```
+
+```{r, echo=FALSE, warning=FALSE, message=FALSE}
+library(admiraldev)
+```
+
+# Visit variables (`AVISIT`, `AVISITN`, `AWLO`, `AWHI`, ...) {#visits}
+
+The most common ways of deriving `AVISIT` and `AVISITN` are:
+
+- The variables are set to the collected visits (`VISIT` and `VISITNUM`).
+- The variables are set based on time windows.
+
+The former one can be done by calling `mutate()` like in the vignettes and the
+template scripts.
+
+For the latter one a (study specific) reference dataset needs to be created
+which provides for each visit the start and end day (`AWLO` and `AWHI`) and the
+values of other visit related variables (`AVISIT`, `AVISITN`, `AWTARGET`, ...).
+```{r}
+windows <- tribble(
+  ~AVISIT,    ~AWLO, ~AWHI, ~AVISITN, ~AWTARGET,
+  "BASELINE",   -30,     1,        0,         1,
+  "WEEK 1",       2,     7,        1,         5,
+  "WEEK 2",       8,    15,        2,        11,
+  "WEEK 3",      16,    22,        3,        19,
+  "WEEK 4",      23,    30,        4,        26
+)
+```
+Then the visits can be assigned based on the analysis day (`ADY`) by calling
+`derive_vars_joined`:
+```{r}
+adbds <- tribble(
+  ~USUBJID, ~ADY,
+  "1",       -33,
+  "1",        -2,
+  "1",         3,
+  "1",        24,
+  "2",        NA,
+)
+
+derive_vars_joined(
+  adbds,
+  dataset_add = windows,
+  filter_join = AWLO <= ADY & ADY <= AWHI
+)
+```
+
+# Period, Subperiod, and Phase Variables
+
+If periods, subperiods, or phases are used, the corresponding variables have to
+be consistent across all datasets. This can be achieved by defining the periods,
+subperiods, or phases once and then use this definition for all datasets. The
+definition can be stored in ADSL or in a separate dataset. In the following this
+separate dataset is called period reference dataset.
+
+## Period Reference Dataset
+
+The period reference dataset contains one observation per subject and period,
+subperiod, or phase. For example:
```

Comment:
"sub-period"?

Reply:
The ADaMIG uses "subperiod". Thus I used the same term.

Reply:
ok